meerkat/columns/instances_column.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import itertools
+from typing import Sequence
+
+from tqdm import tqdm
+
+from meerkat.columns.list_column import ListColumn
+from meerkat.tools.lazy_loader import LazyLoader
+
+ops = LazyLoader("torchvision.ops")
+
+
+class InstancesColumn(ListColumn):
+    def __init__(self, data: Sequence = None, *args, **kwargs):
+
+        super(InstancesColumn, self).__init__(data=data, *args, **kwargs)
+
+    def num_instances(
+        self, batch_size: int = 32
+    ) -> ListColumn:  # TODO(Priya): ListColumn or some other type
+        # Returns a column with the number of instances in each Instances object
+
+        data = []  # Holds the number of instances in each object
+        for batch in tqdm(
+            self.batch(batch_size),
+            total=(len(self) // batch_size + int(len(self) % batch_size != 0)),
+        ):
+            batch_data = [len(instance) for instance in batch]
+            data = list(itertools.chain(data, batch_data))
+
+        data_col = ListColumn(data)
+
+        return data_col
+
+    def get_field(self, field: str, batch_size: int = 32) -> ListColumn:
+        # Returns a column of the required field from each Instances object
+
+        data = []
+        for batch in tqdm(
+            self.batch(batch_size),
+            total=(len(self) // batch_size + int(len(self) % batch_size != 0)),
+        ):
+            batch_data = [instance.get_fields()[field] for instance in batch]
+            data = list(itertools.chain(data, batch_data))
+
+        data_col = ListColumn(data)
+
+        return data_col
+
+    def get_nms(
```

Comment:
It could be useful if NMS should return a new `InstancesColumn` where the boxes that have been suppressed are removed from each instance. The indices may not mean much at the `InstancesColumn` level b/c it'd be hard for the user to make sense of indices that aren't completely visible to them

Reply:
I am returning a ListColumn of boxes tensors now